### PR TITLE
#1760: Media Gallery Page opened successfully if "Enhanced Media Gallery" disabled

### LINKED ISSUE #1760: Media Gallery Page opened successfully if "Enhanced Media Gallery" disabled
### Preconditions (*)

1. Install Magento 2.4-develop with Adobe Stock Integration and IMS
2. Enhanced Media Gallery Disabled 
### Steps to reproduce (*)

1. Open `https://yourdomain.com/admin/media_gallery/media/index` link

### Expected result (*)

404 page

### Actual result (*)

Pages opened successfully
Peek 2020-08-17 18-30

### PR DESCRIPTION
**Pull request by jmonteros422**
### Description (*)

This PR will fixes issue magento/adobe-stock-integration#1760

### Fixed Issues (if relevant)

1. Fixes magento/adobe-stock-integration#1760: Media Gallery page and Category grid page opened successfully if "Enhanced Media Gallery" disabled 

### Preconditions (*)

1. Install Magento 2.4-develop with Adobe Stock Integration and IMS
2. Enhanced Media Gallery Disabled 
### Steps to reproduce (*)

1. Open `https://yourdomain.com/admin/media_gallery/media/index` link

### Expected result (*)

404 page

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by jmonteros422** [comment]
@magento run all tests

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Thanks for the PR @jmonteros422 ! Can you please revert the changes for the Category grid page, only the Media Gallery page should not be acccesible.

Can you please also cover the changes by MFTF tests: 404 page should be returned trying to access media gallery by URL if it's disabled

**Comment by jmonteros422** [comment]
@magento run all tests

File: app/code/Magento/MediaGalleryUi/Test/Mftf/Test/AdminStandaloneMediaGalleryDisabledTest.xml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    <test name="AdminMediaGalleryDisabledPageTest">
+        <annotations>
+            <features value="MediaGallery"/>
+            <useCaseId value="https://github.com/magento/adobe-stock-integration/issues/1760"/>
+            <title value="Standalone Media Gallery Page should return 404 if Media Gallery is disabled"/>
+            <stories value="#1760 Media Gallery Page opened successfully if Enhanced Media Gallery disabled"/>
+            <testCaseId value="https://studio.cucumber.io/projects/131313/test-plan/folders/1337102/scenarios/5106786"/>
+            <description value="Standalone Media Gallery Page should return 404 if Media Gallery is disabled"/>
+            <severity value="CRITICAL"/>
+            <group value="media_gallery_ui"/>
```
Comment (sivaschenko) [inline app/code/Magento/MediaGalleryUi/Test/Mftf/Test/AdminStandaloneMediaGalleryDisabledTest.xml]: Please intorduce a new `media_gallery_ui_disabled` suite that will not enable the new media gallery (that suite does not need any before/after actions actually)

File: app/code/Magento/MediaGalleryUi/Test/Mftf/ActionGroup/AssertAdminStandaloneMediaGalleryPageIs404ActionGroup.xml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+ /**
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+
+<actionGroups xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
```
Comment (sivaschenko) [inline app/code/Magento/MediaGalleryUi/Test/Mftf/ActionGroup/AssertAdminStandaloneMediaGalleryPageIs404ActionGroup.xml]: Please introduce this action group as `app/code/Magento/Backend/Test/Mftf/ActionGroup/AssertAdminPageIs404ActionGroup.xml`

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Thanks for the updates @jmonteros422 ! Please see my comments

**Comment by jmonteros422** [comment]
@magento run all tests

File: app/code/Magento/MediaGalleryUi/Test/Mftf/Test/AdminStandaloneMediaGalleryDisabledTest.xml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+            <title value="Standalone Media Gallery Page should return 404 if Media Gallery is disabled"/>
+            <stories value="#1760 Media Gallery Page opened successfully if Enhanced Media Gallery disabled"/>
+            <testCaseId value="https://studio.cucumber.io/projects/131313/test-plan/folders/1337102/scenarios/5106786"/>
+            <description value="Standalone Media Gallery Page should return 404 if Media Gallery is disabled"/>
+            <severity value="CRITICAL"/>
+            <group value="media_gallery_ui_disabled"/>
+        </annotations>
+        <before>
+            <actionGroup ref="AdminLoginActionGroup" stepKey="loginAsAdmin"/>
+            <actionGroup ref="AdminMediaGalleryEnhancedEnableActionGroup" stepKey="disableEnhancedMediaGallery"/>
```
Comment (sivaschenko) [inline app/code/Magento/MediaGalleryUi/Test/Mftf/Test/AdminStandaloneMediaGalleryDisabledTest.xml]: There is no need to disable media gallery as it is disabled by default

File: app/code/Magento/MediaGalleryUi/Test/Mftf/Test/AdminStandaloneMediaGalleryDisabledTest.xml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+            <actionGroup ref="AdminOpenStandaloneMediaGalleryActionGroup" stepKey="openStandaloneMediaGallery"/>
```
Comment (sivaschenko) [inline app/code/Magento/MediaGalleryUi/Test/Mftf/Test/AdminStandaloneMediaGalleryDisabledTest.xml]: opening media gallery is actually part of the test, not the precondition

File: app/code/Magento/MediaGalleryUi/Test/Mftf/ActionGroup/AssertAdminPageIs404ActionGroup.xml [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+<!--
+ /**
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+
+<actionGroups xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+              xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/actionGroupSchema.xsd">
+    <actionGroup name="AssertAdminPageIs404ActionGroup">
```
Comment (sivaschenko) [inline app/code/Magento/MediaGalleryUi/Test/Mftf/ActionGroup/AssertAdminPageIs404ActionGroup.xml]: please move the action group to Backend module (see the exact path in the comment from previous review)

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Thanks for updates @jmonteros422 ! Please see my comments

**Comment by jmonteros422** [comment]
@magento run all tests

**Comment by sivaschenko** [comment]
@magento run Functional Tests EE

**Review by sivaschenko — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @sivaschenko, thank you for the review.
[ENGCOM-8064](https://jira.corp.magento.com/browse/ENGCOM-8064) has been created to process this Pull Request

**Comment by chalov-anton** [comment]
:heavy_check_mark: QA Passed

If thee _New Media Gallery_ is **disabled**, the https://yourdomain.com/admin/media_gallery/media/index link leads to _404 Page not found_

Manual testing scenario - https://studio.cucumber.io/projects/131313/test-plan/folders/1337102/scenarios/5106786

**Pull request merged**